fishtank/src/node.ts

```diff
@@ -54,6 +58,42 @@ export class Node {
     return sdk.connectRpc(false, true) as Promise<RpcSocketClient>
   }
 
+  async getNodeStatus(): Promise<'started' | 'stopped' | 'error'> {
+    let rpc
+    try {
+      rpc = await this.connectRpc()
+    } catch {
+      return 'stopped'
+    }
+    try {
+      const status = await rpc.node.getStatus()
+      return status.content.node.status
+    } catch {
+      return 'error'
+    } finally {
+      rpc.close()
+    }
+  }
+
+  async isStarted(): Promise<boolean> {
+    return (await this.getNodeStatus()) === 'started'
+  }
+
+  async waitForStart(options?: { timeout?: number }): Promise<void> {
+    let stop = false
+    const timeout = options?.timeout ?? DEFAULT_WAIT_TIMEOUT
+    setTimeout(() => {
```

Comment:
you should use this return value and use the cancel token in the return or else this will be a hanging promise in the event loop after this function returns.

It's a harmless thing, just a small nit.

Reply:
Good point, done